### DISCUSSION TITLE
Default route parameters are not populated in Laravel redirects

### DISCUSSION THREAD

**Discussion by tiagocpeixoto** [tighten/ziggy#700] [Q&A]
### Ziggy version

1.8.1

### Laravel version

10.38.2

### Description

I defined a Laravel route based on a tenant prefix:

```
Route::get('/{tenant}/dashboard', [DashboardController::class, 'show'])->name('dashboard');
```

And then I created a middleware to redirect to the correct tenant if the user enters one that they are not authorized to access. The middleware does something like this:

```
URL::defaults(['tenant' => $tenantId]);
return redirect()->route('dashboard');
```

When the redirect is complete, the React SPA throws the following exception:
`error: Ziggy error: 'tenant' parameter is required for route 'dashboard'.`

But if a change the redirect to:

```
URL::defaults(['tenant' => $tenantId]);
Inertia::location(route('dashboard'));
```

then no error occurs.

In the React client, the link is:

```
<Link href={route("dashboard")}>Users</>
```

It seems that redirects with `Inertia::location` update the Ziggy state (???) and correctly populate the default parameters, but this is not the case with `redirect()->route(...)`.

IF the redirect is done with `redirect()->route('dashboard')`, the workaround on the React side is to explicitly set the expected param:
```
<Link href={route("dashboard", {tenant})}>Users</>
```

In this case, the `tenant` param was obtained from the `usePage` hook **(Ziggys defaults !!!)**, which demonstrates that the default (tenant) parameter is being loaded on the client:
```
const { ziggy: { defaults: { tenant } } } = usePage().props;
```

So is it a bug or a misunderstanding in using the default parameters?

Thanks in advance for your help and thanks again for the wonderful package!

### Ziggy call and context

```js
As explained in the description.
```

### Ziggy configuration

```js
As explained in the description.
```

### Route definition

```php
As explained in the description.
```

**Comment by bakerkretzmar** [comment]
For this particular edge case, a tenant accessing the wrong subdomain, I would suggest keeping `Inertia::location(route('dashboard'))` because that's by far the simplest and cleanest solution. `Inertia::location()` actually forces your app to _not_ do an Inertia request/response and to instead perform a full page reload, which is exactly what you want here because it will re-render the `@routes` Blade directive and update the Ziggy config. You _can_ update the Ziggy config dynamically without reloading the page but I think that's overkill for this use case, assuming that this kind of incorrect access and redirect will be pretty uncommon.
